source/qb64pe.bas

```diff
@@ -25599,7 +25599,12 @@ FUNCTION EvalPreIF (text$, err$)
             END IF
             IF INSTR(symbol$, "=") THEN 'check to see if we're equal in any case with =
                 UserFound = 0
+                IF l$ = UserDefine(0, 7) THEN 'we're comparing VERSION numbers
```

Comment:
I mentioned it on Discord but this would be a good candidate for some [tests](https://github.com/QB64-Phoenix-Edition/QB64pe/blob/main/docs/testing.md#qb64-based-tests). They get a little interesting since the version can't be assumed to be constant (the tests will get run with new versions), but it should still be somewhat straight forward to do some basic tests, Ex:

```vb
$LET verCopy = VERSION

$IF VERSION < 4.0.0 THEN
Print "4.0.0 major comparison: Success"
$ELSE
Print "4.0.0 major comparison: Failure"
$END IF

$IF 3.1.0 < VERSION THEN
Print "3.1.0 minor comparison: Success"
$ELSE
Print "3.1.0 minor comparison: Failure"
$END IF

$IF VERSION = verCopy THEN
Print "Equal comparison: Success"
$ELSE
Print "Equal comparison: Failure"
$END IF

' ...
```

Also test the error cases (which IMO should produce a compilation error, which we can also write tests for via an `.err` file):

```vb
' Should produce a "Cannot compare 'asdfasdf' as a version number" error
$IF asdfasdf = VERSION THEN
Print "Failure"
$END IF
```